### PR TITLE
Fix(#16459) xml parse regression

### LINKED ISSUE #16459: Regression in parsing XML symbols used in old if-else syntax
## Compiler version
3.2.1

## Minimized code

```Scala
//> using lib "org.scala-lang.modules::scala-xml:2.1.0"

@main def Test() = {
  val elem = if(true) {
    <script type="text/javascript">
      'location.reload()'
      'foo bar'
    </script>
  } else <div>empty</div>

  println(elem)
}
```

Replacing parentheses with new if/then syntax allows to compile this code
```scala
// works with both `if true then ...` and `if true then { ...` 
//> using lib "org.scala-lang.modules::scala-xml:2.1.0"

@main def Test() = {
  val elem = if true then {
    <script type="text/javascript">
      'location.reload()'
      'foo bar'
    </script>
  } else <div>empty</div>

  println(elem)
}
```

## Output

```scala
[error] ./test.sc:7:25: unclosed character literal
[error]       'location.reload()'
[error]      
[error] ./test.sc:8:15: unclosed character literal
[error]       'foo bar'
[error]     
```

## Expectation
Should compile with output: 
```
<script type="text/javascript">
      'location.reload()'
    </script>

```

### PR DESCRIPTION
**Pull request by i10416**
close #16459 

The parser could not parse `if expr` that contains single-quoted text(s) inside XML literal with newline(s) because `followedByToken`, which is used to detect `do` or `then` token after `if`, unintentionally consumed XMLSTART symbol, which prevented the parser from delegating parse to XML parser.

### CODE REVIEW

**Comment by i10416** [comment]
scala-xml/shared/src/test/scala/scala/xml/XMLTest.scala:90:24 

```
 [E032] Syntax Error: /__w/dotty/dotty/community-build/community-projects/scala-xml/shared/src/test/scala/scala/xml/XMLTest.scala:90:24 
[error] 90 |      val actual = for (case t @ <book><title>Blabla</title></book> <- NodeSeq.fromSeq(books.child).toList)
[error]    |                        ^^^^
[error]    |                        pattern expected
[error]    |
[error]    | longer explanation available when compiling with `-explain`
[error] -- [E040] Syntax Error: /__w/dotty/dotty/community-build/community-projects/scala-xml/shared/src/test/scala/scala/xml/XMLTest.scala:91:8 
[error] 91 |        yield t
[error]    |        ^^^^^
[error]    |        '<-' expected, but 'yield' found
[error] -- [E040] Syntax Error: /__w/dotty/dotty/community-build/community-projects/scala-xml/shared/src/test/scala/scala/xml/XMLTest.scala:94:4 
[error] 94 |    }
[error]    |    ^
[error]    |    '<-' expected, but '}' found
[error] -- [E007] Type Mismatch Error: /__w/dotty/dotty/community-build/community-projects/scala-xml/shared/src/test/scala/scala/xml/XMLTest.scala:92:0 
[error] 92 |        yield t
[error]    |               ^
[error]    |Found:    (??? : => Nothing)
[error]    |Required: ?{ foreach: ? }
[error]    |Note that implicit conversions were not tried because the result of an implicit conversion
[error]    |must be more specific than ?{ foreach: [applied to (
[error]    |  x$1 =>
[error]    |    x$1 match 
[error]    |      {
[error]    |        case (_root_.scala.Predef.???) =>
[error]    |          _root_.scala.Predef.???.foreach(
[error]    |            x$1 =>
[error]    |              x$1 match 
[error]    |                {
[error]    |                  case _root_.scala.Predef.??? =>
[error]    |                    _root_.scala.Predef.???.foreach(
[error]    |                      x$1 =>
[error]    |                        x$1 match 
[error]    |                          {
[error]    |                            case assertEquals(expected, actual) =>
[error]    |                              _root_.scala.Predef.???
[error]    |                          }
[error]    |                    )
[error]    |                }
[error]    |          )
[error]    |      }
[error]    |) returning <?>] }
[error]    |
[error]    | longer explanation available when compiling with `-explain`
```

File: compiler/src/dotty/tools/dotc/parsing/Parsers.scala [hunk 1]
```diff
@@ -906,6 +906,7 @@ object Parsers {
       var braces = 0
       while (true) {
         val token = lookahead.token
+        if (query != LARROW && token == XMLSTART) return false
```
Comment (i10416) [inline compiler/src/dotty/tools/dotc/parsing/Parsers.scala:909]: This fixes the following case,

```scala
  val casePatMatch = for (case t @ <foo>FooBar</foo> <- Seq(xml))
   yield t
```

but it still fails with the following case.

```scala
  val casePatMatchWithCond = for (case t @ <foo>FooBar</foo> <- Seq(xml))
   yield t
```

I need to investigate routines more.

Comment (i10416) [reply compiler/src/dotty/tools/dotc/parsing/Parsers.scala:909]: Ah, I made a mistake.

```diff
- val casePatMatchWithCond = for (case t @ <foo>FooBar</foo> if true <- Seq(xml))
+ val casePatMatchWithCond = for (case t @ <foo>FooBar</foo><- Seq(xml) if true )
   yield t
```

File: tests/run/i16459.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  
+  // Note: This passes in Scala  2.12.18 and 2.13.12 too.
+  val auxiliary4 = if(true) <div attr="...">A</div>else <div attr="...">B</div>
+
+  // Pattern match without guard.
+  // Note: This passes in Scala  2.12.18 and 2.13.12 too.
+  val auxiliary5 = for (case _ @ <div>empty</div> <- Seq(xml)) yield ()
+  // Note: These pass in Scala  2.12.18 and 2.13.12.
+  val auxiliary6 = for (case _ @ <div>empty</div><- Seq(xml)) yield ()
+  val auxiliary7 = for (case _ @ <div>empty</div><-Seq(xml)) yield ()
```
Comment (i10416) [inline tests/run/i16459.scala:53]: The left arrow just after xml closing tag breaks the parser.

File: tests/run/i16459.scala [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  // Pattern match with if guard.
+  // Note: This passes in Scala  2.12.18 and 2.13.12 too.
+  val auxiliary8 = for (case _ @ <foo>FooBar</foo> <- Seq(xml) if true)
+  // Note: These pass in Scala  2.12.18 and 2.13.12.
+  val auxiliary9 = for (case _ @ <foo>FooBar</foo><- Seq(xml) if true)
+  val auxiliary10 = for (case _ @ <foo>FooBar</foo><-Seq(xml) if true)
+   yield ()
```
Comment (i10416) [inline tests/run/i16459.scala:60]: same as https://github.com/lampepfl/dotty/pull/19531/files#r1466874075

File: tests/run/i16459.scala (hunk 2 above)
Comment (i10416) [reply tests/run/i16459.scala:53]: Fixed

**Review by odersky — APPROVED** [review]
LGT, thanks for the fix!

**Pull request merged**